### PR TITLE
Add Status F08 <-> C/F conversion routines

### PR DESCRIPTION
**Pull request by dalcinl**

### CODE REVIEW

File: include/mpi.h [hunk 1]
```diff
@@ -160,6 +160,10 @@ enum {
     MPI_F_STATUS_SIZE   = 8
 };
 
+typedef struct {
+  int internal[MPI_F_STATUS_SIZE];
```
Comment (dalcinl) [inline include/mpi.h:164]: Alternatively, we could just do:

```suggestion
  MPI_Status mpi_val;
```

which would be more alike to the way opaque handles are defined.

File: mpi.h [hunk 2]
```diff
@@ -160,6 +160,10 @@ enum {
     MPI_F_STATUS_SIZE   = 8
 };
 
+typedef struct {
+  int MPI_internal[MPI_F_STATUS_SIZE];
```
Comment (dalcinl) [inline mpi.h:164]: Alternatively, we could just do:

```suggestion
  MPI_Status mpi_val;
```

which would be more alike to the way MPI types are defined within F08.

**Comment by dalcinl** [comment]
@jeffhammond @hzhou Gentle reminder to review this PR. Please look at my question about how to define `MPI_F08_status`.

**Comment by jeffhammond** [comment]
You need to use MPI_Fint not int, in anything that's compatible with Fortran.

**Comment by dalcinl** [comment]
> You need to use MPI_Fint not int, in anything that's compatible with Fortran.

Obviously... :face_with_head_bandage:. Fixed now, I settled with `MPI_Fint mpi_val[MPI_F_STATUS_SIZE]`.

**Review by jeffhammond — APPROVED** [review]

**Pull request merged**